woocommerce/woocommerce
# Show the payment method as text if there's no `brand` or `last4` digits are present

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR changes how we render the payment method. Before, when there was no brand or card info we showed a generic card icon, which does not make a lot of sense if the payment was `cash`, for example.

With this PR, we check if `brand` and `last4` are empty and, in that case, we render the title of the payment method instead of the generic card icon.

More context in this comment https://github.com/woocommerce/woocommerce/pull/52399#issuecomment-2449397636



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install `WooPayments` and enable sandbox mode.
2. In `wp-admin/admin.php?page=wc-settings&tab=checkout` make sure you have enabled the `Cash on delivery` payment method.
3. Make a purchase with a fake card and another one with `Cash on delivery` (or any other method different than card).
4. Go to `WooCommerce > Order` and note the numbers of the two orders you just did.
5. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
6. Make the following request for the two order numbers you got on step 4 request using the consumer key and secret generated in the previous step:
```bash 
curl --insecure -u [KEY]:[SECRET] --request POST --url 'https://[STORE_URL]/wp-json/wc/v3/orders/[ORDER_NUMBER]/receipt?force_new=true' --header 'Content-Type: application/json'
```
7. Visit the URL from the response and make sure the order made with the card shows the card icon and the last 4 digits of the card. Also, ensure the order made with cash (or the other method) shows the name of the method in text (no icon).

| Cash | Card | Check |
|--------|--------|--------|
| <img width="445" alt="Screenshot 2024-11-05 at 12 29 08" src="https://github.com/user-attachments/assets/09ff9b4d-b747-41d0-8a1e-13eb1d6d4530">| <img width="446" alt="Screenshot 2024-11-05 at 12 29 16" src="https://github.com/user-attachments/assets/17553b07-a439-4593-b426-a9bc67267ea3">| <img width="448" alt="Screenshot 2024-11-05 at 12 29 13" src="https://github.com/user-attachments/assets/1e55914c-a9f7-4712-9951-6f483d21083e">| 




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Show the payment method as text if there's no brand or last4 digits are present.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
